02-peer-protocol.md

```diff
@@ -1437,6 +1456,454 @@ a large feerate change, instead sets their `sats` to zero, and decline to
+We allow both nodes to initiate RBF, because any one of them may want to take
+this opportunity to splice additional funds into or out of the channel without
+waiting for the initial splice transaction to confirm.
+
+Since splice transactions always spend the current channel funding output, the
+RBF attempts automatically double-spend each other.
+
+#### The `tx_ack_rbf` Message
+
+##### Requirements
+
+The sending node:
+  - MAY set `funding_output_contribution` to a different value than the
+    `funding_contribution_satoshis` used in `splice_init` or `splice_ack`,
+    or in previous RBF attempts.
+
+The receiving node:
+  - If `funding_output_contribution` is negative and its absolute value is
+    greater than the sending node's current channel balance:
+    - MUST send a `warning` and close the connection or send an `error`
+      and fail the channel.
+
+### Splice Completion
+
+Once splice transactions have been signed but haven't reached acceptable
+depth, channel operations go back to normal and HTLCs can be exchanged,
+with the caveat that payments must be valid for all splice transactions.
+
+Nodes keep track of multiple commitment transactions (one for the current
+funding transaction and one for each splice transaction) and exchange
+signatures for each of these commitment transactions.
+
+```
++------------+        +-----------+
+| Funding Tx |---+--->| Commit Tx |
++------------+   |    +-----------+
+                 |    +-----------+            +-----------+
+                 +--->| Splice Tx |----------->| Commit Tx |
+                 |    +-----------+            +-----------+
+                 |    +---------------+        +-----------+
+                 +--->| Splice RBF #1 |------->| Commit Tx |
+                 |    +---------------+        +-----------+
+                 |    +---------------+        +-----------+
+                 +--->| Splice RBF #2 |------->| Commit Tx |
+                      +---------------+        +-----------+
+```
+
+The splice completes by exchanging `splice_locked` messages, at which point
+the locked transaction replaces the previous funding transaction.
+
+#### The `splice_locked` Message
+
+1. type: 77 (`splice_locked`)
+2. data:
+   * [`channel_id`:`channel_id`]
+   * [`sha256`:`splice_txid`]
+
+##### Requirements
+
+Each node:
+  - If any splice transaction reaches acceptable depth:
+    - MUST send `splice_locked`.
+
+Once a node has sent and received `splice_locked`:
+  - MUST consider the locked splice transaction to be the new funding
+    transaction for all future `commitment_signed` messages and splice
+    negotiations.
+  - SHOULD discard the previous funding transaction and RBF attempts.
+  - MUST send `announcement_signatures` with `short_channel_id` matching
+    the locked splice transaction.
+
+On reconnection:
+  - MUST retransmit its last `splice_locked` if the `commitment_number`
+    is the same as before sending `splice_locked`.
```

Comment:
This should only be sent if we receive a `next_funding_txid` matches our current channel funding txid (ie we have `splice_locked` but our peer hasn't).

Probably we should drop this clause as the reestablish `splice_locked` clause already covers this.

Reply:
No? I think we're not on the same page here regarding `splice_locked` or the inclusion of the `next_funding_txid` field, it's important to clarify this. If we've correctly exchanged signatures, we won't include `next_funding_txid` in `channel_reestablish` at all. But we may not have exchanged `splice_locked` at that point.

If I then send `splice_locked` but we disconnect, I'll need to retransmit it on reconnection. The only stage at which I know I don't need to retransmit it is if we signed a new commitment after locking.